packages/autocomplete-core/src/__tests__/createConcurrentSafePromise.test.ts

```diff
@@ -0,0 +1,41 @@
+import { createConcurrentSafePromise } from '../createConcurrentSafePromise';
+
+function defer<TValue>(fn: () => TValue, timeout: number) {
+  return new Promise<TValue>((resolve) => {
+    setTimeout(() => resolve(fn()), timeout);
+  });
+}
+
+describe('createConcurrentSafePromise', () => {
+  test('resolves the value from the last call', async () => {
```

Comment:
another test where they resolve in order too maybe?

Reply:
Done in 9fc796bce65fdb894f1770239254c5ab3d5f3942.